collector.js

```diff
@@ -98,18 +114,31 @@ class AlAzureCollector {
         // Somebody can still pass 'undefined' as hostmetaElems and it will be added to args list.
         var hm = hostmetaElems ? hostmetaElems : this._defaultHostmetaElems();
         var ingestc = this._ingestc;
-        var stats = this._collectionStats;
         
         if (messages && messages.length > 0) {
-            alcollector.AlLog.buildPayload(this._hostId, this._sourceId, hm, messages, formatFun, function(err, payload){
+            let buildPayloadObj = {
+                hostId: this._hostId, 
+                sourceId: this._sourceId, 
+                hostmetaElems: hm, 
+                content: messages, 
+                parseCallback: formatFun, 
+                filterJson: this._filterJson, 
+                filterRegexp: this._filterRegex
+            };
+            alcollector.AlLog.buildPayload(buildPayloadObj, (err, data) => {
                 if (err) {
                     return callback(err);
                 } else {
-                    ingestc.sendAicspmsgs(payload)
+                    ingestc.sendLogmsgs(data.payload)
                         .then( resp => {
-                            // Subtract 2 bytes in order not to count array brackets [].
-                            const bytes = JSON.stringify(messages).length - 2;
-                            return stats.putLogStats(bytes, messages.length, callback);
+                            let lmcStats = this._prepareLmcStats(data.raw_count, data.raw_bytes);
+                            ingestc.sendLmcstats(JSON.stringify([lmcStats]))
+                                .then(resp => {
+                                    return callback(null, resp);
+                                })
+                                .catch(exception => {
+                                    return callback(exception);
```

Comment:
please return `callback(null)` here as well. We should not fail the invocation if we fail to update stats. It may result in log recollection.

Reply:
Done